src/swell/tasks/link_geos_output.py

```diff
@@ -53,72 +51,161 @@ def execute(self) -> None:
+        # destination file name
+        for state in states:
+            date = dt.strptime(state['date'], datetime_formats['iso_format'])
+            src = self.forecast_dir('his_' + date.strftime('%Y_%m_%d_%H') + '.nc')
+            dst = state['ocn_filename']
+            self.src_dst_dict[src] = dst
+
+        # Add the background (beginning of the window) file
+        src = self.forecast_dir('his_' + self.bkgr_time_dto.strftime('%Y_%m_%d_%H') + '.nc')
+        dst = 'MOM6.res.' + self.bkgr_time_iso + '.nc'
+
+        # Append dst to the dictionary
+        self.src_dst_dict[src] = dst
+
+    # ----------------------------------------------------------------------------------------------
+
+    def link_mom6_history_3d(self) -> Tuple[str, str]:
+        # Using a 3D window and hence the background is a single file, in the
+        # middle of the DA window
+        src = self.forecast_dir('his_' + self.bkgr_time_dto.strftime('%Y_%m_%d_%H') + '.nc')
+        dst = 'MOM6.res.' + self.bkgr_time_iso + '.nc'
+
+        # Append dst to the dictionary
+        self.src_dst_dict[src] = dst
+
+    # ----------------------------------------------------------------------------------------------
+
+    def cice6_history_formatter(self,
+                                src_date: dt,
+                                hour_prefix: str,
+                                ) -> str:
+
+        # Extract the date part and calculate seconds for the src_history format
+        date_str = src_date.strftime('%Y-%m-%d')
+        seconds = src_date.hour * 3600 + src_date.minute * 60 + src_date.second
+
+        return self.forecast_dir(f'iceh_{hour_prefix}.{date_str}-{seconds}.nc')
 
     # ----------------------------------------------------------------------------------------------
 
-    def link_mom6_history(self, src_dst_dict):
+    def cice6_history_hour_prefix(self,
+                                  background_frequency: str = 'PT3H',
+                                  ) -> str:
+        # There is no background_frequency for 3D window so we set PT3H
+        # as the default. This is defined in CICE6 input file, ice_in
+        # -------------------------------------------------------------------
+        # TODO: A safer alternative would be to get this information from ice_in (nml)
+        # but the format is not consistent and varies according to output frequency
+        # histfreq       = 'h','x','x','x','x'
+        # histfreq_n     =  3 , 1 , 1 , 1 , 1
 
-        # Create links to GEOS history for SOCA inputs
+        # Convert background frequency to 02d format
+        duration = isodate.parse_duration(background_frequency)
+        hist_hours, remainder = divmod(duration.total_seconds(), 60*60)
+
+        return f'{int(hist_hours):02d}h'
+
+    # ----------------------------------------------------------------------------------------------
+
+    def link_cice6_history_4d(self) -> Tuple[str, str]:
```

Comment:
This has no `return` statement, so it returns `None`, _not_ a Tuple.

```suggestion
    def link_cice6_history_4d(self) -> None:
```

Reply:
oops..